src/poller.ts

```diff
@@ -1,35 +1,134 @@
+import {
+  DEFAULT_INITIAL_CONFIG_REQUEST_RETRIES,
+  DEFAULT_POLL_CONFIG_REQUEST_RETRIES,
+  POLL_JITTER_PCT,
+} from './constants';
+
 export interface IPoller {
   start: () => Promise<void>;
   stop: () => void;
 }
 
 export default function initPoller(
-  interval: number,
+  intervalMs: number,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   callback: () => Promise<any>,
+  options?: {
+    maxPollRetries?: number;
+    maxStartRetries?: number;
+    errorOnFailedStart?: boolean;
+    pollAfterFailedStart?: boolean;
+  },
 ): IPoller {
   let stopped = false;
+  let failedAttempts = 0;
+  let nextPollMs = intervalMs;
+
+  const start = async () => {
+    stopped = false;
+    let startRequestSuccess = false;
+    let startAttemptsRemaining =
+      1 + (options?.maxStartRetries ?? DEFAULT_INITIAL_CONFIG_REQUEST_RETRIES);
+
+    let startError = null;
+
+    while (!startRequestSuccess && startAttemptsRemaining > 0) {
+      try {
+        await callback();
+        startRequestSuccess = true;
+      } catch (pollingError) {
+        console.warn(
+          `Eppo SDK encountered an error with initial poll of configurations: ${pollingError.message}`,
+        );
+        if (--startAttemptsRemaining > 0) {
+          const jitterMs = randomJitterMs(intervalMs);
+          console.warn(
+            `Eppo SDK will retry the initial poll again in ${jitterMs} ms (${startAttemptsRemaining} attempts remaining)`,
+          );
+          await new Promise((resolve) => setTimeout(resolve, jitterMs));
```

Comment:
This is effectively a sleep, and that way `await`ing start will keep waiting for retries.